buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -190,15 +190,14 @@ ByteBuf allocate(int size, int maxCapacity) {
             boolean willCleanupFastThreadLocals = FastThreadLocalThread.willCleanupFastThreadLocals(currentThread);
             AdaptiveByteBuf buf = AdaptiveByteBuf.newInstance(willCleanupFastThreadLocals);
             try {
-                AdaptiveByteBuf result = allocate(size, maxCapacity, currentThread, buf);
-                if (result != null) {
-                    // The ownership of the buf is now part of the result that we return.
+                if (allocate(size, maxCapacity, currentThread, buf)) {
+                    // Allocation did work, the ownership will is transferred to the caller.
                     buf = null;
-                    return result;
+                    return buf;
```

Comment:
Tests are failing because we set `buf` to `null` here.

Reply:
ups